### PR TITLE
Add TFDebertaV2ForMultipleChoice

### LINKED ISSUE #25537: ValueError: Unrecognized configuration class <class>
### System Info

```
transformer v. 4.31.0'
````

### Who can help?

@sgugger

### Information

- [ ] The official example scripts
- [X] My own modified scripts

### Tasks

- [X] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

I tried to load `'microsoft/deberta-v3-large'` using AutoModelForMultipleChoice and TFAutoModelForMultipleChoice class. But I got error with TF* class.

```python
from transformers import AutoModelForMultipleChoice
from transformers import TFAutoModelForMultipleChoice

deberta_v3_large = 'microsoft/deberta-v3-large'

# OK
torch_model = AutoModelForMultipleChoice.from_pretrained(deberta_v3_large)

# NOT OK
tf_model = TFAutoModelForMultipleChoice.from_pretrained(deberta_v3_large)
```

Errro logs

```
╭─────────────────────────────── Traceback (most recent call last) ────────────────────────────────╮
│ in <module>:1                                                                                    │
│                                                                                                  │
│ ❱ 1 tf_model = TFAutoModelForMultipleChoice.from_pretrained(deberta_v3_large)                    │
│   2                                                                                              │
│                                                                                                  │
│ /opt/conda/lib/python3.10/site-packages/transformers/models/auto/auto_factory.py:496 in          │
│ from_pretrained                                                                                  │
│                                                                                                  │
│   493 │   │   │   return model_class.from_pretrained(                                            │
│   494 │   │   │   │   pretrained_model_name_or_path, *model_args, config=config, **hub_kwargs,   │
│   495 │   │   │   )                                                                              │
│ ❱ 496 │   │   raise ValueError(                                                                  │
│   497 │   │   │   f"Unrecognized configuration class {config.__class__} for this kind of AutoM   │
│   498 │   │   │   f"Model type should be one of {', '.join(c.__name__ for c in cls._model_mapp   │
│   499 │   │   )                                                                                  │
╰──────────────────────────────────────────────────────────────────────────────────────────────────╯
ValueError: Unrecognized configuration class <class 
'transformers.models.deberta_v2.configuration_deberta_v2.DebertaV2Config'> for this kind of AutoModel: 
TFAutoModelForMultipleChoice.
Model type should be one of AlbertConfig, BertConfig, CamembertConfig, ConvBertConfig, DistilBertConfig, 
ElectraConfig, FlaubertConfig, FunnelConfig, LongformerConfig, MobileBertConfig, MPNetConfig, RemBertConfig, 
RobertaConfig, RobertaPreLayerNormConfig, RoFormerConfig, XLMConfig, XLMRobertaConfig, XLNetConfig.
```

### Expected behavior

They should work in same manner!

### PR DESCRIPTION
**Pull request by raghavanone**
# What does this PR do?

Fixes #25537 

Add TFDebertaV2ForMultipleChoice model

@ArthurZucker

### CODE REVIEW

**Comment by amyeroberts** [comment]
cc @Rocketknight1

File: src/transformers/models/deberta_v2/modeling_tf_deberta_v2.py [hunk 1]
```diff
@@ -1529,3 +1529,102 @@ def call(
             hidden_states=outputs.hidden_states,
             attentions=outputs.attentions,
         )
+
+
+@add_start_docstrings(
+    """
+    Bert Model with a multiple choice classification head on top (a linear layer on top of the pooled output and a
```
Comment (Rocketknight1) [inline src/transformers/models/deberta_v2/modeling_tf_deberta_v2.py:1536]:
```suggestion
    DeBERTa v2 Model with a multiple choice classification head on top (a linear layer on top of the pooled output and a
```

**Review by Rocketknight1 — APPROVED** [review]
Looks good to me! Seems like a clean port of the BERT code to DeBERTa-v2. One nitpick where a docstring wasn't changed, but other than that I think this is ready to merge!

**Comment by Rocketknight1** [comment]
That test is an issue with the CI rather than this PR, you can ignore it! Are you ready for me to merge now?

**Comment by raghavanone** [comment]
> That test is an issue with the CI rather than this PR, you can ignore it! Are you ready for me to merge now?

Yes

**Comment by Rocketknight1** [comment]
Ugh, it won't let me merge, which means we'll need to rebase to get the tests working. Can you:

1) [Pull the latest changes to your fork's main branch](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/syncing-a-fork)
2) In your local working repo, `git checkout main` and `git pull` to ensure your local main branch is up to date
3) In your local repo, `git checkout fix_issue_25537` and then `git rebase main` to rebase onto the latest `main` branch
4) Finally, `git push --force` to upload the rebased branch to Github

After that, tests should pass!

**Comment by raghavanone** [comment]
> Ugh, it won't let me merge, which means we'll need to rebase to get the tests working. Can you:
> 
> 1. [Pull the latest changes to your fork's main branch](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/syncing-a-fork)
> 2. In your local working repo, `git checkout main` and `git pull` to ensure your local main branch is up to date
> 3. In your local repo, `git checkout fix_issue_25537` and then `git pull` and `git rebase main` to rebase onto the latest `main` branch
> 4. Finally, `git push --force` to upload the rebased branch to Github
> 
> After that, tests should pass!

Done, Lets hope the test passes.

**Comment by Rocketknight1** [comment]
Looks like everything's passing now - sorry about the CI issues, and thanks for a very clean and useful PR!

**Pull request merged**